phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexToolIT.java

```diff
@@ -237,6 +237,80 @@ public void testWithSetNull() throws Exception {
         }
     }
 
+    @Test
+    public void testIncorrectRebuild() throws Exception {
+        // This test is for building non-transactional mutable global indexes with direct api
+        if (localIndex || transactional || !mutable || !directApi || useSnapshot) {
+            return;
+        }
+        String schemaName = generateUniqueName();
+        String dataTableName = generateUniqueName();
+        String dataTableFullName = SchemaUtil.getTableName(schemaName, dataTableName);
+        String indexTableName = generateUniqueName();
+        String indexTableFullName = SchemaUtil.getTableName(schemaName, indexTableName);
+        Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        try (Connection conn = DriverManager.getConnection(getUrl(), props)) {
+            conn.setAutoCommit(true);
+            conn.createStatement().execute("create table " + dataTableFullName +
+                    " (id varchar(10) not null primary key, val1 varchar(10), val2 varchar(10), val3 varchar(10)) COLUMN_ENCODED_BYTES=0");
+            conn.createStatement().execute("CREATE INDEX " + indexTableName + " on " +
+                    dataTableFullName + " (val1) include (val2, val3)" );
+            //insert a full row
+            conn.createStatement().execute("upsert into " + dataTableFullName + " values ('a', 'ab', 'efgh', 'abcd')");
+            Thread.sleep(1000);
```

Comment:
Good to use the EnvironmentEdgeManager rather than thread.sleep. It's both faster and more deterministic.

Reply:
I dont really need to sleep here. I can skip this.